addons/website_slides/static/src/js/slides_course_fullscreen_player.js

```diff
@@ -7,413 +7,585 @@ odoo.define('website_slides.fullscreen', function (require) {
+     *
+     * The widget will trigger an event `change_slide` when the video is at
+     * its end, and `slide_completed` when the player is at 30 sec before the
+     * end of the video (30 sec before is considered as completed).
+     */
+    var VideoPlayer = Widget.extend({
+        template: 'website.slides.fullscreen.video',
+        youtubeUrl: 'https://www.youtube.com/iframe_api',
 
-    tag.src = "https://www.youtube.com/iframe_api";
-    var firstScriptTag = document.getElementsByTagName('script')[0];
-    firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
-
-
-    var Fullscreen = Widget.extend({
-        custom_events: {
-            quiz_next_slide: '_goToNextSlide',
-            quiz_completed: '_onQuizCompleted'
-        },
-        /**
-        * @override
-        * @param {Object} el
-        * @param {Object} data holding channelId and optionally upload and publish control parameters
-        */
-        init: function (el, courseId, slideId, userId) {
-            this.courseID = parseInt(courseId, 10);
-            this.slideID = parseInt(slideId, 10);
-            this.userID = parseInt(userId, 10);
-            this.course = undefined;
-            this.slide = undefined;
-            this.slides = [];
-            this.nextSlide = undefined;
-            this.previousSlide = undefined;
-            this.url = undefined;
-            this.urlToSmallScreen = undefined;
-            this.activetab = undefined;
-            this.player = undefined;
-            this.goToQuiz = false;
-            this.slideTitle = undefined;
-            return this._super.apply(this,arguments);
+        init: function (parent, slide) {
+            this.slide = slide;
+            return this._super.apply(this, arguments);
         },
         start: function (){
-            var self = this;
-            this.url = window.location.pathname;
-            this.urlToSmallScreen = this.url.replace('/fullscreen','');
-            this._getSlides();
-            this._renderPlayer();
-            this._bindListEvents();
-            $(document).keydown(this._onKeyDown.bind(this));
-            return this._super.apply(this, arguments);
+            return $.when(this._super.apply(this, arguments), this._loadYoutubeAPI());
         },
-        //--------------------------------------------------------------------------
-        // Private
-        //--------------------------------------------------------------------------
-         /**
-         * @private
-         * Renders the player accordingly to the current slide
-         */
-        _renderPlayer: function (){
-            var self = this;
-
+        _loadYoutubeAPI: function () {
             var def = $.Deferred();
-            if (this.slide.slide_type === 'webpage') {
-                this._fetchHtmlContent().then(function () {
-                    self._renderWebpage();
+            if(!document.querySelector('script[src="' + this.youtubeUrl + '"]')) {
```

Comment:
Can't this be jqueryed?
```
$('script...')
$('<script/>', {src: this.youtubeUrl})
...

```

Reply:
It can be, but we need the `onload` callback. To do so, in jquery, we will need a `.done()`, and i'm not sure it will be compatible with the change of promise in master.
Moreover, in the web editor, some script are loaded dynamically (also in PDFjs lib) and they are all using native  `createElement`.

Reply:
According to this: https://stackoverflow.com/questions/4638296/jquery-equivalent-of-body-onload
It would more likely be:

```
if($(document).find('script[src="' + this.youtubeUrl + '"]').length === 0) {
    $youtubeElement = $('<script/>, {src: this.youtubeUrl}).load(function() {
        def.resolve();
    });
    $(document.head).append($youtubeElement);
}
```

It looks a little sexier and it avoids mixing native JS dom manipulation with jQuery stuff in the same file. But if it's too much chipotage just leave it like that.

Reply:
I did not find the `.load` ! Thanks, it will do the job !